examples/ogbn_train.py

```diff
@@ -0,0 +1,266 @@
+        y = batch.y[:batch.batch_size].squeeze().to(torch.long)
+        loss = F.cross_entropy(out, y)
+        loss.backward()
+        optimizer.step()
+
+        total_loss += float(loss)
+        total_correct += int(out.argmax(dim=-1).eq(y).sum())
+        pbar.update(batch.batch_size)
+
+    pbar.close()
+    loss = total_loss / len(train_loader)
+    approx_acc = total_correct / split_idx['train'].size(0)
+    return loss, approx_acc
+
+
+@torch.no_grad()
+def test(loader: NeighborLoader, val_steps=None) -> float:
+    model.eval()
+
+    total_correct = total_examples = 0
+    for i, batch in enumerate(loader):
+        if val_steps is not None and i >= val_steps:
+            break
+        batch = batch.to(device)
+        batch_size = batch.num_sampled_nodes[0]
+        out = model(batch.x, batch.edge_index)[:batch_size]
+        pred = out.argmax(dim=-1)
+        y = batch.y[:batch_size].view(-1).to(torch.long)
+
+        total_correct += int((pred == y).sum())
+        total_examples += y.size(0)
+
+    return total_correct / total_examples
+
+
+if args.use_gat:
+    from torch_geometric.nn.models import GAT
+    model = GAT(
+        in_channels=dataset.num_features,
+        hidden_channels=num_hidden_channels,
+        num_layers=num_layers,
+        out_channels=dataset.num_classes,
+        dropout=args.dropout,
+        heads=args.num_heads,
+    )
+else:
+    from torch_geometric.nn.models import GraphSAGE
+    model = GraphSAGE(
+        in_channels=dataset.num_features,
+        hidden_channels=num_hidden_channels,
+        num_layers=num_layers,
+        out_channels=dataset.num_classes,
+        dropout=args.dropout,
+    )
+
+model = model.to(device)
+optimizer = torch.optim.Adam(model.parameters(), lr=args.lr,
+                             weight_decay=args.wd)
+
+if verbose:
+    prep_time = round(time.perf_counter() - wall_clock_start, 2)
+    print("Total time before training begins (prep_time)=", prep_time,
+          "seconds")
+    print("Training...")
+
+test_accs = []
+val_accs = []
+times = []
+train_times = []
+inference_times = []
+best_val = best_test = 0.
+start = time.time()
+
+model.reset_parameters()
+optimizer = torch.optim.Adam(model.parameters(), lr=args.lr)
+
+for epoch in range(1, num_epochs + 1):
+    train_start = time.time()
+    loss, _ = train(epoch)
+    train_end = time.time()
+    train_times.append(train_end - train_start)
+
+    inference_start = time.time()
+    val_acc = test(val_loader)
+    test_acc = test(test_loader)
+
+    inference_times.append(time.time() - inference_start)
+    test_accs.append(test_acc)
+    val_accs.append(val_acc)
+    print(f'Epoch {epoch:02d}, Loss: {loss:.4f}, ',
+          f'Train Time: {train_end - train_start:.4f}s')
+    print(f'Val: {val_acc * 100.0:.2f}%,', f'Test: {test_acc * 100.0:.2f}%')
+
+    if val_acc > best_val:
+        best_val = val_acc
+    if test_acc > best_test:
+        best_test = test_acc
```

Comment:
Related to my above comment: I don't understand why it's ok to pick the best metric evaluated on the test set across all epochs.

Reply:
well this was just for tracking it seems. there is no checkpointing being done but yeah the test set should only be run once at the end anyways. @guanxingithub please remove the test acc tracking since it should only be run once at the end and reported then

Reply:
Best metric evaluated on the test set is one of indexes to make statistical accuracy analysis. With all the information we collected during training, we can compare the performance and accuracy of particular model among different platforms, among different version of tools used to run training.

Reply:
@guanxingithub in general in machine learning it is not proper to run the test set before training/val has finished.
Tracking the best val acc across multiple epochs for perflab is fine, but there should only be one final test accuracy for them to track. running test during training is sort of "cheating". please address me and akihiro's concerns in order to get this merged.

Reply:
Hmmm, still not sure why it makes sense to evaluate the model on the test set during training. Just to make sure, I'm suggesting deleting L233 and L241-L242.

Reply:
Done